### PR TITLE
[GS][W-8529083] Add 'Page Type' field to section metadata

### PR DESCRIPTION
**Pull request by fparodi-salesforce**
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata
- Custom Field
  - GetStartedChecklistSection__mdt.Belongs_to__c

# Deleted Metadata

### CODE REVIEW

**Comment by fparodi-salesforce** [comment]
[W-8529083](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008kekeIAA/view)

File: src/objects/GetStartedChecklistSection__mdt.object [hunk 1]
```diff
@@ -1,6 +1,32 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
     <description>NPSP Get Started Page sections</description>
+    <fields>
+        <fullName>Belongs_to__c</fullName>
```
Comment (force2b) [inline src/objects/GetStartedChecklistSection__mdt.object:5]: How about the following?
- Label: "Page Type"
- Api Name: Page_Type__c
- Help Text: The type of page (Admin or End User) that the options in this GetStarted Section will be rendered in.
- Description: (same as help text)

What do you think @smarovich?

File: src/objects/GetStartedChecklistSection__mdt.object [hunk 2]
```diff
@@ -39,6 +65,6 @@
         <unique>false</unique>
     </fields>
     <label>NPSP Get Started Checklist Section</label>
-    <pluralLabel>NPSP Get Started Checklist Sections</pluralLabel>
+    <pluralLabel>NPSP Get Started Checklist Section</pluralLabel>
```
Comment (force2b) [inline src/objects/GetStartedChecklistSection__mdt.object:68]: Did you remove the "s" from the plural label? If not, can you put it back in?
(If you agree @smarovich)

File: src/lwc/gsResources/gsResources.js-meta.xml [hunk 3]
```diff
@@ -4,6 +4,11 @@
     <isExposed>true</isExposed>
     <masterLabel>NPSP Get Started Page Resources</masterLabel>
     <targets>
-          <target>lightning__AppPage</target>
+        <target>lightning__AppPage</target>
     </targets>
+    <targetConfigs>
+        <targetConfig targets="lightning__AppPage">
+            <property name="belongsTo" label="Belongs to" type="String" datasource="Admin,End User" />
```
Comment (force2b) [inline src/lwc/gsResources/gsResources.js-meta.xml:11]:
```suggestion
            <property name="pageType" label="Page Type" type="String" datasource="Admin,End User" />
```

File: src/lwc/gsResources/gsResources.js [hunk 4]
```diff
@@ -17,6 +17,11 @@ import gsResourcesPartnersLink from '@salesforce/label/c.gsResourcesItemPartners
 
 export default class gsResources extends LightningElement {
 
+    /**
+    * @description To select which data display in this component
+    */
+    @api belongsTo = 'Admin';
```
Comment (force2b) [inline src/lwc/gsResources/gsResources.js:23]:
```suggestion
    @api pageType = 'Admin';
```

File: src/lwc/gsChecklistSetup/gsChecklistSetup.js-meta.xml [hunk 5]
```diff
@@ -6,4 +6,9 @@
     <targets>
         <target>lightning__AppPage</target>
     </targets>
+    <targetConfigs>
+        <targetConfig targets="lightning__AppPage">
+            <property name="belongsTo" label="Belongs to" type="String" datasource="Admin,End User" />
```
Comment (force2b) [inline src/lwc/gsChecklistSetup/gsChecklistSetup.js-meta.xml:11]:
```suggestion
            <property name="pageType" label="Page Type" type="String" datasource="Admin,End User" />
```

File: src/lwc/gsChecklistSetup/gsChecklistSetup.js [hunk 6]
```diff
@@ -13,8 +13,13 @@ export default class gsChecklistSetup extends LightningElement {
...
     @track checklists = [];
     
+    /**
+    * @description To select which data display in this component
+    */
+    @api belongsTo = 'Admin';
+
     connectedCallback() {
-        getChecklists()
+        getChecklists({'belongTo': this.belongsTo})
```
Comment (force2b) [inline src/lwc/gsChecklistSetup/gsChecklistSetup.js:22]:
```suggestion
        getChecklists({'pageType': this.pageType})
```

**Review by force2b — COMMENTED** [review]
Mostly a question for @smarovich about the field name (and as a result the related property/var names). If Sarah agrees, then all the instances of "belongsTo", "Belongs To", etc. would be changed appropriately.

File: src/objects/GetStartedChecklistSection__mdt.object (hunk 1 above)
Comment (smarovich) [reply src/objects/GetStartedChecklistSection__mdt.object:5]: @force2b How about "The type of Get Started page (Admin or End User) where these sections will appear."?

File: src/objects/GetStartedChecklistSection__mdt.object (hunk 2 above)
Comment (smarovich) [reply src/objects/GetStartedChecklistSection__mdt.object:68]: Agreed!

Comment (force2b) [inline src/objects/GetStartedChecklistSection__mdt.object:68]:
```suggestion
    <pluralLabel>NPSP Get Started Checklist Sections</pluralLabel>
```

File: src/classes/GS_ChecklistSetup.cls [hunk 7]
```diff
@@ -43,9 +43,9 @@ public with sharing class GS_ChecklistSetup {
      * @return List<ChecklistSection>
      */
     @AuraEnabled
-    public static List<ChecklistSection> getChecklists() {
+    public static List<ChecklistSection> getChecklists(String belongTo) {
         storedItemsId = getMarkedAsCompletedCheckListItemByUser();
-        List<GetStartedChecklistSection__mdt> checklistSections = getSectionsAndSubsections();
+        List<GetStartedChecklistSection__mdt> checklistSections = getSectionsAndSubsections(belongTo);
```
Comment (force2b) [inline src/classes/GS_ChecklistSetup.cls:48]:
```suggestion
        List<GetStartedChecklistSection__mdt> checklistSections = getSectionsAndSubsections(pageType);
```

File: src/classes/GS_ChecklistSetup.cls [hunk 8]
```diff
@@ -129,7 +129,7 @@ public with sharing class GS_ChecklistSetup {
      * @return List<GetStartedChecklistSection__mdt>
      * @note [ORDER BY Position__c ASC] doesn't work well in the inner query, I need to sort outside SOQL
      */
-    private static List<GetStartedChecklistSection__mdt> getSectionsAndSubsections() {
+    private static List<GetStartedChecklistSection__mdt> getSectionsAndSubsections(String belongTo) {
```
Comment (force2b) [inline src/classes/GS_ChecklistSetup.cls:132]:
```suggestion
    private static List<GetStartedChecklistSection__mdt> getSectionsAndSubsections(String pageType) {
```

File: src/classes/GS_ChecklistSetup.cls [hunk 9]
```diff
@@ -140,6 +140,7 @@ public with sharing class GS_ChecklistSetup {
                     FROM GetStartedChecklistItems__r
                 )
             FROM GetStartedChecklistSection__mdt
+            WHERE Belongs_to__c = :belongTo
```
Comment (force2b) [inline src/classes/GS_ChecklistSetup.cls:143]:
```suggestion
            WHERE Page_Type__c = :belongTo
```

File: src/objects/GetStartedChecklistSection__mdt.object [hunk 10]
```diff
@@ -1,6 +1,32 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
     <description>NPSP Get Started Page sections</description>
+    <fields>
+        <fullName>Belongs_to__c</fullName>
+        <description>To select in which checklist has displayed this section</description>
+        <externalId>false</externalId>
+        <fieldManageability>DeveloperControlled</fieldManageability>
+        <inlineHelpText>To select in which checklist has displayed this section</inlineHelpText>
+        <label>Belongs to</label>
```
Comment (force2b) [inline src/objects/GetStartedChecklistSection__mdt.object:10]:
```suggestion
        <fullName>Page_Type__c</fullName>
        <description>The type of Get Started page (Admin or End User) where these sections will appear.</description>
        <externalId>false</externalId>
        <fieldManageability>DeveloperControlled</fieldManageability>
        <inlineHelpText>The type of Get Started page (Admin or End User) where these sections will appear.</inlineHelpText>
        <label>Page Type</label>
```

File: src/flexipages/GetStarted.flexipage [hunk 11]
```diff
@@ -5,6 +5,10 @@
             <componentName>gsVideoHeader</componentName>
         </componentInstances>
         <componentInstances>
+            <componentInstanceProperties>
+                <name>belongsTo</name>
```
Comment (force2b) [inline src/flexipages/GetStarted.flexipage:9]:
```suggestion
                <name>pageType</name>
```

File: src/flexipages/GetStarted.flexipage [hunk 12]
```diff
@@ -19,6 +23,10 @@
     </flexiPageRegions>
     <flexiPageRegions>
         <componentInstances>
+            <componentInstanceProperties>
+                <name>belongsTo</name>
```
Comment (force2b) [inline src/flexipages/GetStarted.flexipage:27]:
```suggestion
                <name>pageType</name>
```

**Review by force2b — CHANGES_REQUESTED** [review]

File: src/objects/GetStartedChecklistSection__mdt.object (hunk 10 above)
Comment (smarovich) [reply src/objects/GetStartedChecklistSection__mdt.object:10]: @fparodi-salesforce I just realized that we need to make one more small change. This field is on a single section, right? I think it should be "The type of Get Started page (Admin or End User) where this section will appear."

File: src/lwc/gsResources/gsResources.js-meta.xml (hunk 3 above)
Comment (smarovich) [reply src/lwc/gsResources/gsResources.js-meta.xml:11]: I agree with Mike's suggestion to change the field name.

**Review by force2b — CHANGES_REQUESTED** [review]
Looking very good, but I think the only file still missing is the updated page layout for the GetStartedChecklistSection object that includes the new Page_Type field

**Review by force2b — APPROVED** [review]

File: src/objectTranslations/GetStartedChecklistSection__mdt-en_US.objectTranslation [hunk 13]
```diff
@@ -13,6 +13,19 @@
         <label><!-- Description Label Name --></label>
         <name>Description_Label__c</name>
     </fields>
+    <fields>
+        <help><!-- To select in which checklist has displayed this section --></help>
```
Comment (smarovich) [inline src/objectTranslations/GetStartedChecklistSection__mdt-en_US.objectTranslation:17]: I recommend changing this to "Determines which checklist will display this section". I recommend changing this for all of the translations.

File: src/objects/GetStartedChecklistSection__mdt.object [hunk 14]
```diff
@@ -13,6 +13,32 @@
         <type>Text</type>
         <unique>false</unique>
     </fields>
+    <fields>
+        <fullName>Page_Type__c</fullName>
+        <description>The type of page (Admin or End User) that the options in this GetStarted Section will be rendered in.</description>
```
Comment (smarovich) [inline src/objects/GetStartedChecklistSection__mdt.object:18]: @fparodi-salesforce Please change this to "The type of Get Started page (Admin or End User) where this section will appear."

File: src/objects/GetStartedChecklistSection__mdt.object [hunk 15]
```diff
@@ -13,6 +13,32 @@
...
+        <externalId>false</externalId>
+        <fieldManageability>DeveloperControlled</fieldManageability>
+        <inlineHelpText>The type of page (Admin or End User) that the options in this GetStarted Section will be rendered in.</inlineHelpText>
```
Comment (smarovich) [inline src/objects/GetStartedChecklistSection__mdt.object:21]: @fparodi-salesforce Please change this to "The type of Get Started page (Admin or End User) where this section will appear."

**Review by smarovich — CHANGES_REQUESTED** [review]

**Review by smarovich — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**